### DISCUSSION TITLE
iOS Notification Images

### DISCUSSION THREAD

**Discussion by websoptimization** [invertase/react-native-firebase#4663] [Q&A] [answered]
I want to integrate push notification with features image in my project. for that i followed all steps which are define [here](https://rnfirebase.io/messaging/ios-notification-images) 

after that i configure appID and provision profile for notificationExtensionService in my apple account

issue is when i send notification from firebase console with image url (which has 178kb size and it's HTTPS), notification will display without image on first time run from xCode and when i run again from xCode and send push notification from firebase console that time notification is coming correctly with image 

So help me at which point i made mistake or something which i missed to integrate

**Comment by mikehardy** [comment]
I'm unsure but per #4225 perhaps @davidepalazzo knows ? I don't use this functionality personally, so I'm no help, sorry

**Comment by davidepalazzo** [comment, marked as answer]
Hi @websoptimization. With the provisioning profile I assume you're referring to an APN auth key as Xcode by default handles the signing for you.

Without knowing the before/after of your project setup, it's challenging to point out a missing step. Notification setup involves a few moving pieces (e.g. APN, Firebase configuration, RNFirbase, code, etc.) that's why in the guide we recommend to start only after Cloud Messaging is properly setup and working.

It could have been anything from latency between Firebase and Apple to a cached Xcode build. Happy to hear you got it working eventually.

**Reply by websoptimization** [reply]
@davidepalazzo thank for reply

I already integrated cloud messaging correctly and it's working fine (it's display push notification with title and body) but the problem is when i run project from xCode first time, didReceiveNotificationRequest method of NotificationExtensionService is not tigger when i send features image notification and that's why that image is not displaying in my notification.

after installed app on my device and then again i run project that time i got features image in my notification. 
So why image is not displaying in first time run and have to run again project?

**Comment by davidepalazzo** [comment]
@websoptimization no worries.

When you say `didReceiveNotificationRequest` isn't firing, how did you debug it? Keep in mind that logs don't get outputted for extensions - you can read more about why [here](https://developer.apple.com/forums/thread/43205).

As mentioned, what you're experiencing can be caused my multiple factors - It's hard to pin point what's going wrong without seeing your full setup - Firebase settings, info.plist, RN version, RNF version, etc.

Sorry I can't be of more help. All the best with it!
